File: src/main/java/com/embabel/guide/domain/drivine/HasDiscordUserInfoData.java
package com.embabel.guide.domain.drivine;

import org.jetbrains.annotations.NotNull;

/**
 * Marker interface for types that contain Discord user information.
 * Provides access to the Discord user data component.
 */
public interface HasDiscordUserInfoData {

    /**
     * Get the Discord user info data
     * @return the Discord user info data
     */
    @NotNull
    DiscordUserInfoData getDiscordUserInfo();
}

File: src/main/kotlin/com/embabel/guide/domain/HasDiscordUserInfoData.kt
package com.embabel.guide.domain

/**
 * Marker interface for types that contain Discord user information.
 * Provides access to the Discord user data component.
 */
interface HasDiscordUserInfoData {

    /**
     * Get the Discord user info data
     * @return the Discord user info data
     */
    val discordUserInfo: DiscordUserInfoData
}
